### PR TITLE
Change YoroiNFT and TokenInfo data structure

### PR DESCRIPTION
**Pull request by michalszorad**
Resolves YOMO-458

### CODE REVIEW

**Comment by michalszorad** [comment]
This is put on hold until we resolve the token info encoding issue YOMO-539 to confirm the new data structure works.

**Comment by michalszorad** [comment]
I'm looking into resolving the conflicts

**Review by banklesss — COMMENTED** [review]
I have a question

Don't we need to define "hybrid" tokens in this case?

File: apps/wallet-mobile/src/features/Send/useCases/ListAmountsToSend/AddToken/SelectTokenFromListScreen.tsx [hunk 1]
```diff
@@ -92,7 +92,9 @@ const NftList = ({canAddAmount}: {canAddAmount: boolean}) => {
   const strings = useStrings()
   const {nfts} = useNfts(wallet)
   const amountsSelected = Object.keys(targets[selectedTargetIndex].entry.amounts)
-  const filteredAndSorted = nfts.filter(filterOutSelected(amountsSelected)).sort((a, b) => sortNfts(a.name, b.name))
+  const filteredAndSorted = nfts
+    .filter(filterOutSelected(amountsSelected))
+    .sort((a, b) => sortNfts(a.name ?? '', b.name ?? ''))
```
Comment (banklesss) [inline apps/wallet-mobile/src/features/Send/useCases/ListAmountsToSend/AddToken/SelectTokenFromListScreen.tsx]: 👍🏻

File: apps/wallet-mobile/src/legacy/format.ts
Comment (jemubm) [inline apps/wallet-mobile/src/legacy/format.ts:1]: if there are useful thing here, should they be moved to another place?

**Comment by michalszorad** [comment]
Good point @jemubm I'll look into that

File: apps/wallet-mobile/src/yoroi-wallets/types/tokens.ts [hunk 2]
```diff
@@ -63,14 +49,47 @@ export type LegacyToken = {
...
 
 export type NFTAsset = {
   key: '721'
   metadata: AssetMetadata
 }
+
+type TokeInfoCommon = {
+  id: string
+  fingerprint: string
+  name?: string
```
Comment (stackchain) [inline apps/wallet-mobile/src/yoroi-wallets/types/tokens.ts]: name can be empty '' but can't be undefined.

**Review by stackchain — CHANGES_REQUESTED** [review]

**Comment by michalszorad** [comment]
@jemubm we'll leave hybrid for now and use only FT/NFT as there are some unknowns and need to be discussed with Ruslan

File: apps/wallet-mobile/src/yoroi-wallets/types/tokens.ts [hunk 3]
```diff
@@ -63,14 +49,36 @@ export type LegacyToken = {
...
+  id: string
+  group: string
+  fingerprint: string
+  image: string | undefined
+  icon: string | undefined
+  decimals: number | undefined
+  symbol: string | undefined
+  name: string
+  description: string | undefined
+  ticker: string | undefined
```
Comment (michalszorad) [inline apps/wallet-mobile/src/yoroi-wallets/types/tokens.ts:71]: @stackchain @jemubm would it make sense to add `author` and `files` properties here for the NFT object? We're accessing these on the NFT detail screen / nft preview

File: apps/wallet-mobile/src/features/Send/common/filterBySearch.ts [hunk 4]
```diff
@@ -5,8 +5,17 @@ export const filterBySearch = (searchTerm: string) => {
...
+    if (tokenInfo.kind === 'ft') {
+      return (
+        tokenInfo.ticker?.toLocaleLowerCase()?.includes(searchTermLowerCase) ||
+        tokenInfo.name?.toLocaleLowerCase()?.includes(searchTermLowerCase)
+      )
+    }
+
+    if (tokenInfo.kind === 'nft') {
+      const name = tokenInfo.name?.toLocaleLowerCase() ?? null
+      if (name) return name.toLowerCase().includes(searchTermLowerCase)
```
Comment (banklesss) [inline apps/wallet-mobile/src/features/Send/common/filterBySearch.ts]: `toLocaleLowerCase ` ?

File: apps/wallet-mobile/src/components/AmountItem/AmountItem.tsx [hunk 5]
```diff
@@ -33,7 +34,7 @@ export const AmountItem = ({wallet, style, amount}: AmountItemProps) => {
 
       <Middle>
         <Text numberOfLines={1} ellipsizeMode="middle" style={styles.name} testID="tokenInfoText">
-          {name}
+          {isEmptyString(name) ? '-' : name}
```
Comment (banklesss) [inline apps/wallet-mobile/src/components/AmountItem/AmountItem.tsx]: i remember @stackchain asked me to add this logic to the top

File: apps/wallet-mobile/src/yoroi-wallets/types/tokens.ts (hunk 2 above)
Comment (michalszorad) [reply apps/wallet-mobile/src/yoroi-wallets/types/tokens.ts]: That's now updated

File: apps/wallet-mobile/src/components/AmountItem/AmountItem.tsx (hunk 5 above)
Comment (michalszorad) [reply apps/wallet-mobile/src/components/AmountItem/AmountItem.tsx]: I've moved this up

File: apps/wallet-mobile/src/features/Send/common/filterBySearch.ts (hunk 4 above)
Comment (michalszorad) [reply apps/wallet-mobile/src/features/Send/common/filterBySearch.ts]: I've removed the toLowerCase, as it was lowercased twice

**Review by banklesss — APPROVED** [review]

**Review by stackchain — APPROVED** [review]

**Review by stackchain — APPROVED** [review]

**Pull request merged**